api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -1635,6 +1636,13 @@ message AccessRequestCreate {
 
   // ResourceNames contains the names of the requested resources.
   repeated string ResourceNames = 17 [(gogoproto.jsontag) = "resource_names,omitempty"];
+
+  // AllowedResourceAccessIDs is the set of resources to which access is being requested,
+  // paired with additional information such as ResourceConstraints.
+  repeated types.ResourceAccessID RequestedResourceAccessIDs = 18 [
```

Comment:
Using the functional type as is in audit logs is dangerous IMO, it means that every field we add or remove directly changes the shape of the audit log event, which is a sort of fuzzy API surface consumed by customers through their SIEM and as such it should be treated with a modicum of care.

For example, if it ever turns out that we have to add new fields with duplicated info while also populating the old ones (for the sake of backwards compatibility) to ResourceAccessID we are going to have duplicated info in the audit log events too.

Reply:
I see. What do you think would be a solution here? Treating ResourceAccessIDs the same as before, regardless of whether they're constrained, and just providing `resource_names`?